### PR TITLE
V2: fix some docstrings

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
As I've read through the code base, I've seen some parts of docstrings that looked incorrect. Some of these are more obvious than others. I've carefully reviewed these changes, but will appreciate another pair of eyes on it before we go to merge.

### CODE REVIEW

File: chemprop/models/model.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def training_step(self, batch: TrainingBatch, batch_idx):
         preds = self.predictor.train_step(Z)
         l = self.criterion(preds, targets, mask, w_s, self.w_t, lt_mask, gt_mask)
 
-        self.log("train/loss", l, prog_bar=True)
+        self.log("train_loss", l, prog_bar=True)
```
Comment (hwpang) [inline chemprop/models/model.py:152]: What does this `self.log` function do?

File: chemprop/nn/agg.py [hunk 2]
```diff
@@ -47,15 +47,15 @@ def forward(self, H: Tensor, batch: Tensor) -> Tensor:
         Parameters
         ----------
         H : Tensor
-            A tensor of shape ``V x d`` containing the batched node-level representations of ``n``
+            A tensor of shape ``V x d`` containing the batched node-level representations of ``b``
```
Comment (hwpang) [inline chemprop/nn/agg.py:50]: Did we define `b` anywhere?

File: chemprop/nn/message_passing/base.py [hunk 3]
```diff
@@ -182,13 +182,11 @@ def forward(self, bmg: BatchMolGraph, V_d: Tensor | None = None) -> Tensor:
             an optional tensor of shape ``V x d_vd`` containing additional descriptors for each atom
             in the batch. These will be concatenated to the learned atomic descriptors and
             transformed before the readout phase.
-            **NOTE**: recall that ``V`` is equal to ``num_atoms + 1``, so ``V_d`` must be 0-padded
-            in the 0th row.
```
Comment (hwpang) [inline chemprop/nn/message_passing/base.py:186]: So are we not doing 0th row padding anymore?

**Review by hwpang — COMMENTED** [review]
Overall LGTM with some questions for my own information

File: chemprop/nn/agg.py (hunk 2 above)
Comment (KnathanM) [reply chemprop/nn/agg.py:50]: It is defined [here](https://github.com/chemprop/chemprop/blob/5aeac9eff6c32ff6416df0ed3a03cad94df22cb3/chemprop/nn/message_passing/base.py#L158) in message_passing base.py. We could define `b` here and probably should, but at the same time we don't define what `V` and `d` are. Additionally there are many docstrings that use variables without defining them each time. I think it would be better to collect all the variable definitions into a single reference like what some textbooks do, but that is fairly big task.

File: chemprop/models/model.py (hunk 1 above)
Comment (KnathanM) [reply chemprop/models/model.py:152]: https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.log

It passes information to the lightning progress bar to print. Right now we print something like `Epoch 0: 100%|███████████████████████████████████████████████████████████████| 1/1 [00:11<00:00, 11.95s/it, v_num=0, train/loss=1.250]`. The validation version uses `val_loss=...` instead as recommended by lightning.

File: chemprop/nn/message_passing/base.py (hunk 3 above)
Comment (KnathanM) [reply chemprop/nn/message_passing/base.py:186]: Yes David told me the zeroth row padding is "a remnant of the old scatter/gather ops in pure torch. With the backend move to `torch_scatter` it's no longer necessary". 
I noticed just now message_passing proto also has this note, so I will add that change.

File: chemprop/models/model.py (hunk 1 above)
Comment (hwpang) [reply chemprop/models/model.py:152]: When we report the metrics, we use `"{val,test}/{metric}"`, so I think it makes sense for it to be `/` here, but instead change `val_loss` to `val/loss`? 
```
        self.log("train/loss", l, prog_bar=True)

        return l

    def validation_step(self, batch: TrainingBatch, batch_idx: int = 0):
        losses = self._evaluate_batch(batch)
        metric2loss = {f"val/{m.alias}": l for m, l in zip(self.metrics, losses)}

        self.log_dict(metric2loss, batch_size=len(batch[0]))
        self.log("val_loss", losses[0], batch_size=len(batch[0]), prog_bar=True)

    def test_step(self, batch: TrainingBatch, batch_idx: int = 0):
        losses = self._evaluate_batch(batch)
        metric2loss = {f"test/{m.alias}": l for m, l in zip(self.metrics, losses)}

        self.log_dict(metric2loss, batch_size=len(batch[0]))
```

**Review by hwpang — APPROVED** [review]
LGTM! will merge once the tests pass.
